pkg/osgi_config_manager.go

```diff
@@ -18,23 +18,24 @@ type OSGiConfigManager struct {
 }
 
 func (cm *OSGiConfigManager) ByPID(pid string) OSGiConfig {
-	return OSGiConfig{manager: cm, pid: pid}
+	if cm.IsFactoryPID(pid) {
+		fpid, alias := cm.SplitFactoryPID(pid)
+		return OSGiConfig{manager: cm, pid: pid, fpid: fpid, alias: alias}
+	} else {
+		return OSGiConfig{manager: cm, pid: pid}
+	}
 }
 
-func (cm *OSGiConfigManager) ByFactoryPID(pid string) OSGiConfig {
-	factoryPid, cid := cm.splitPID(pid)
-	return OSGiConfig{manager: cm, pid: pid, fpid: factoryPid, cid: cid}
+func (cm *OSGiConfigManager) IsFactoryPID(pid string) bool {
+	return strings.Contains(pid, osgi.ConfigAliasSeparator)
 }
 
-func (cm *OSGiConfigManager) splitPID(pid string) (string, string) {
-	tokens := strings.SplitN(pid, "~", 2)
-	if len(tokens) > 1 {
-		return tokens[0], tokens[1]
-	}
-	if len(tokens) == 1 {
-		return tokens[0], ""
+func (cm *OSGiConfigManager) SplitFactoryPID(pid string) (string, string) {
+	parts := strings.Split(pid, osgi.ConfigAliasSeparator)
+	if len(parts) != 2 {
+		log.Fatalf("cannot parse OSGi config factory PID '%s'", pid)
 	}
-	return "", ""
+	return parts[0], parts[1]
```

Comment:
So in case when `len(parts) == 1` we log the message and then proceed to panic?

Reply:
no, before we have now `if` which does not invoke this method then at all;

the code is more strict